src/main/java/com/epam/page/object/generator/validators/AbstractValidator.java

```diff
@@ -1,129 +1,53 @@
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Set;
+import com.epam.page.object.generator.models.searchrule.CommonSearchRule;
+import com.epam.page.object.generator.models.searchrule.ComplexInnerSearchRule;
+import com.epam.page.object.generator.models.searchrule.ComplexSearchRule;
+import com.epam.page.object.generator.models.searchrule.FormInnerSearchRule;
+import com.epam.page.object.generator.models.searchrule.FormSearchRule;
+import com.epam.page.object.generator.models.webgroup.CommonWebElementGroup;
+import com.epam.page.object.generator.models.webgroup.ComplexWebElementGroup;
+import com.epam.page.object.generator.models.webgroup.FormWebElementGroup;
 
-/**
- * AbstractValidator is the abstract base class for crating a new Validator, which can be used into
- * ValidatorsStarter. <p> For creating a new Validator you need: <ol> <li>Create a new class which
- * extends from {@link AbstractValidator} and override the main method {@link
- * AbstractValidator#isValid(SearchRule, ValidationContext)}.</li> <li>Create default constructor
- * with {@link AbstractValidator#priority}.</li> <li>Override {@link
- * AbstractValidator#getExceptionMessage(SearchRule searchRule, ValidationContext
- * validationContext)}.</li> </ol>
- */
-public abstract class AbstractValidator implements Validator {
+public abstract class AbstractValidator implements ValidatorVisitor {
 
-    private Set<SearchRuleType> supportedSearchRuleTypes;
-
-    /**
-     * Validator priority is a number which specifies the order of executing validators. All
-     * validators which present by the default have some default priority, but if you create your
-     * own validator you must not forget to set priority for you validator into default constructor
-     * and change priories for all default validators if you need. <br/><br/> For priority you need
-     * to use int numbers: <br/> <ul> <li>0-49 fot validators which validate format of JSON
-     * files</li> <li>51+ for validators which validate SearchRules by the urls</li> </ul> For
-     * example: <br/> UniquenessLocatorValidator can have priority equals 51, because it checks that
-     * the SearchRule is uniqueness on the WebPage by the url. It can be like this:<br/> {@code
-     * public UniquenessLocatorValidator() { super(51); } }
-     */
-    private int priority;
-
-    /**
-     * Each validator can work in two modes. <br/> <ul> <li>isValidateAllSearchRules = true<br/> The
-     * validator validates all SearchRules every time.</li> <li>isValidateAllSearchRules = false
-     * (default)<br/> The validator validates only SearchRules that passed the previous
-     * validator.</li> </ul>
-     */
-    private boolean isValidateAllSearchRules = false;
-
-    public AbstractValidator(int priority) {
-        this.priority = priority;
+    @Override
+    public ValidationResult visit(CommonSearchRule commonSearchRule) {
+        return new ValidationResult(true, this + " passed!");
```

Comment:
Use Java 8 default methods and remove abstract class.

Reply:
Done